### PR TITLE
Add usage for containerized conu

### LINKED ISSUE #127: document how to use containerized conu

### PR DESCRIPTION
**Pull request by lachmanfrantisek**
Signed-off-by: lachmanfrantisek <flachman@redhat.com>

Resolves #127

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,25 @@ $ dnf install python{2,3}-conu conu-doc
...
+## Docker container
+
+You can try conu also in the container, but you have to mount docker socket and use `--privileged` option:
+
+```
+docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock:z --privileged modularitycontainers/conu:0.2.0 python3
+```
+
+```python
+from conu import DockerBackend
```
Comment (dhodovsk) [inline README.md]: This confused me a bit. You could add one explanation line - that this is some example and also add python interpreter prompt `>>>`.

File: README.md [hunk 2]
```diff
@@ -28,6 +28,25 @@ $ dnf install python{2,3}-conu conu-doc
...
+:
```
Comment (dhodovsk) [inline README.md]: Here could be a comment `# Your python code` or `# Here you can play with conu` or some simple  conu example, e.g the helper function `get_container_output` introduced in https://github.com/fedora-modularity/conu/pull/154

File: README.md [hunk 3]
```diff
@@ -28,6 +28,25 @@ $ dnf install python{2,3}-conu conu-doc
 
 Please visit [our documentation](http://conu.readthedocs.io/en/latest/installation.html) for more info on installation.
 
+## Docker container
+
+You can try conu also in the container, but you have to mount docker socket and use `--privileged` option:
+
+```
+docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock:z --privileged modularitycontainers/conu:0.2.0 python3
```
Comment (dhodovsk) [inline README.md]: Running in non-privileged mode: `docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock:z modularitycontainers/conu:0.2.0 make test-doc-examples` went OK in my system - aside from atomic mount example which always need superuser. I wonder what could make a difference. IMO we should investigate this more.

Comment (lachmanfrantisek) [reply README.md]: Maybe @TomasTomecek should know that?

File: README.md (hunk 1 above)
Comment (lachmanfrantisek) [reply README.md]: I've added better example -- should I remove output/logs?

File: README.md (hunk 3 above)
Comment (TomasTomecek) [reply README.md]:
```
$ man 2 mount
...
Appropriate privilege (Linux: the CAP_SYS_ADMIN capability) is required to mount filesystems.
...
```
+1 to Dominika, I would then just mention, that if they require to mount containers/images; they need to supply `--cap-add SYS_ADMIN`

Comment (lachmanfrantisek) [reply README.md]: Ok, thanks.

Comment (lachmanfrantisek) [reply README.md]: There is probably something bad on my site, but it does not work for me..;-(

Comment (lachmanfrantisek) [reply README.md]: I've only changed  `--privileged` to `--cap-add SYS_ADMIN`. Is it ok?

Comment (lachmanfrantisek) [reply README.md]: @TomasTomecek  I've made it work when the selinux is turned off...;-)

Comment (dhodovsk) [reply README.md]: I found this: https://danwalsh.livejournal.com/74095.html. This documentation PR needs a note explaining the way to deal with the issue you experienced.

Comment (lachmanfrantisek) [reply README.md]: Ok, I've got it. Is the new version ok?

File: README.md [hunk 4]
```diff
@@ -28,6 +28,42 @@ $ dnf install python{2,3}-conu conu-doc
...
+
+```python
+>>> from conu import DockerBackend
+>>> backend = DockerBackend()
+11:52:13.022 backend.py        INFO   conu has initiated, welcome to the party!
+>>> image = backend.ImageClass('docker.io/library/nginx')
+11:52:32.562 __init__.py       INFO   docker environment info: ...
+>>> container = image.run_via_binary()
+11:52:51.910 image.py          INFO   run container via binary in background
+72b9566a17763d6154111cdb0acc4224cb12b256c4655b77ad28689a1ecb511f
```
Comment (dhodovsk) [inline README.md]: Is it just me or this is for some reason red?

Comment (lachmanfrantisek) [reply README.md]: Yes, also for me. Do I have to remove that or leave it as is?

Comment (TomasTomecek) [reply README.md]: let's remove it

Comment (dhodovsk) [reply README.md]: It might be caused by the python type of block.

Comment (lachmanfrantisek) [reply README.md]: Yes, it's due to python formatting. I've just removed that line.

**Comment by lachmanfrantisek** [comment]
Just one git-question:

Can I `merge` directly or do I have to `rebase` first?

Thanks

**Comment by TomasTomecek** [comment]
Yup, squashing commits related to #127 into a single one is a good idea so we have compact history. Unrelated commits can be left out separately so it's clear when some flake/typo/whitespace/docs issue was fixed.

> Can I merge directly or do I have to rebase first?

It's better to squash commits first, and then rebase so you wouldn't go insane.

**Comment by lachmanfrantisek** [comment]
Squashed and rebased.

Can I ask once again for the approval?

**Comment by lachmanfrantisek** [comment]
Thanks

**Pull request merged**